test/parallel/test-fs-whatwg-url.js

```diff
@@ -72,3 +72,208 @@ if (common.isWindows) {
     message: `File URL host must be "localhost" or empty on ${os.platform()}`
   }));
 }
+
+// path ending with forward slash are not permitted
+fs.readFile(new URL('file:///c:/tmp/test/'), common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+fs.readFile(new URL('file:///tmp/test/'), common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+common.expectsError(
+  () => {
+    fs.readFileSync(new URL('file:///c:/tmp/test/'));
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+
+common.expectsError(
+  () => {
+    fs.readFileSync(new URL('file:///tmp/test/'));
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+
+//fs.open
+fs.open(new URL('file:///c:/tmp/test/'), common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+fs.open(new URL('file:///tmp/test/'), common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+common.expectsError(
+  () => {
+    fs.openSync(new URL('file:///c:/tmp/test/'));
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+
+common.expectsError(
+  () => {
+    fs.openSync(new URL('file:///tmp/test/'));
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+
+
+//append file
+fs.appendFile(new URL('file:///c:/tmp/test/'), 'test data', common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+fs.appendFile(new URL('file:///tmp/test/'), 'test data', common.expectsError({
+  code: 'ERR_INVALID_FILE_URL_PATH',
+  type: Error,
+  message: 'File URL path illegal operation on a directory'
+}));
+
+common.expectsError(
+  () => {
+    fs.appendFileSync(new URL('file:///c:/tmp/test/'), 'test data');
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+
+common.expectsError(
+  () => {
+    fs.appendFileSync(new URL('file:///tmp/test/'), 'test data');
+  },
+  {
+    code: 'ERR_INVALID_FILE_URL_PATH',
+    type: Error,
+    message: 'File URL path illegal operation on a directory'
+  });
+//creatReadStream
+common.expectsError(
+  () => {
+    fs.createReadStream(new URL('file:///c:/tmp/test/'), 'test data');
```

Comment:
The second argument of `fs.createReadStream` is an option object or an encoding, I am pretty sure this would result in an error. We can leave it as `fs.createReadStream(new URL('file:///c:/tmp/test/'))` here. The same goes to `fs.createWriteStream` as well